pages/common/pydocstyle.md

```diff
@@ -0,0 +1,32 @@
+# pydocstyle
+
+> Statically check Python scripts for compliance with Python docstring conventions.
+> More information: <https://www.pydocstyle.org/en/latest/>.
+
+- Analyze a Python script or all the Python scripts in a specific directory:
+
+`pydocstyle {{file.py|path/to/directory}}`
+
+- Show an [e]xplanation of each error:
+
+`pydocstyle {{-e|--explain}} {{file.py|path/to/directory}}`
+
+- Show [d]ebug information:
+
+`pydocstyle {{-d|--debug}} {{file.py|path/to/directory}}`
+
+- Display the total number of errors:
+
+`pydocstyle --count {{file.py|path/to/directory}}`
+
+- Use a specific configuration file:
+
+`pydocstyle --config {{path/to/config_file}} {{file.py|path/to/directory}}`
+
+- Ignore one or more errors:
+
+`pydocstyle --ignore {{D101,D2,D107,...}} {{file.py|path/to/directory}}`
+
+- Check for errors from a specific convention:
+
+`pydocstyle --convention {{pep257|numpy|google}} {{file.py|directory}}`
```

Comment:
```suggestion
`pydocstyle --convention {{pep257|numpy|google}} {{file.py|/path/to/directory}}`
```

Reply:
```suggestion
`pydocstyle --convention {{pep257|numpy|google}} {{file.py|path/to/directory}}`
```
Is the path absolute? If not, we should remove the starting slash character

Reply:
Can be both AFAIK

Reply:
Yeah, I also think it can be both. From the style guide:
> - When the path cannot be relative,
>   but has to start at the root of the filesystem,
>   prefix it with a slash,
>   such as `get {{/path/to/remote_file}}`.

Reply:
I still do not understand why  we should use relative pathnames in this scenario, maybe because its a use case that suits the tool? (e.g. scanning a file from the source directory)

Reply:
Hello @vitorhcl,

It should be as your suggestion (in order to follow the examples).

Reply:
@tricantivu 

I am using a "relative" path instead of an "aboslute" path becuase my examples trying to explain that `pydocstyle` can be used also at the "project" context.

```
.
├── README.md
├── app
│   ├── producer.py
│   └── requirements.txt
├── setup.cfg
└── tests
    ├── consumer_test.py
    ├── domain_a
    └── domain_b
        ├── class
        │   ├── data.py
        │   └── terminal.py
        └── translator
            └── translator.py
```

```
# Analyze only the `tests/domain_b/` directory.
pydocstyle tests/domain_b/`
```

Is this answer your question?

Reply:
Yeah, otherwise we would need to specify like $HOME/project_path/tests/domain_b/

Reply:
At the end, both suggestion can live together, but I preffer to keep using the previous one (not `/` at the start) in order to have the same example syntax.

Reply:
> @tricantivu 
> 
> 
> 
> I am using a "relative" path instead of an "aboslute" path becuase my examples trying to explain that `pydocstyle` can be used also at the "project" context.
> 
> 
> 
> ```
> 
> .
> 
> ├── README.md
> 
> ├── app
> 
> │   ├── producer.py
> 
> │   └── requirements.txt
> 
> ├── setup.cfg
> 
> └── tests
> 
>     ├── consumer_test.py
> 
>     ├── domain_a
> 
>     └── domain_b
> 
>         ├── class
> 
>         │   ├── data.py
> 
>         │   └── terminal.py
> 
>         └── translator
> 
>             └── translator.py
> 
> ```
> 
> 
> 
> ```
> 
> # Analyze only the `tests/domain_b/` directory.
> 
> pydocstyle tests/domain_b/`
> 
> ```
> 
> 
> 
> Is this answer your question?

Yes, it does.